components/places/src/history_sync/store.rs

```diff
@@ -101,16 +115,15 @@ impl<'a> HistoryStore<'a> {
         root_sync_key: &KeyBundle,
         sync_ping: &mut telemetry::SyncTelemetryPing,
     ) -> Result<()> {
-        let global_state: Cell<Option<String>> = Cell::new(self.get_global_state()?);
+        self.migrate_global_state()?;
         let result = sync_multiple(
             &[self],
-            &global_state,
+            &Cell::new(None),
```

Comment:
The sync manager will need some more thought, but I ended up changing from what I was thinking before:

* There's an `Option<String>` we expect the app to persist for us, which the app treats as opaque. It is used only to populate meta/global when we arrive and find it missing. This is *not* used to change the state of an existing meta/global - the fact that this happens to hold declined engines is really just a coincidence. This part isn't much different conceptually from what we have today (although we now persist far far less)

* There'll be a new API for changing the declined state of an engine. This will take engine names etc, but it will almost certainly take the same, opaque `&mut Option<String>` passed to the sync functions - so this function will both update meta/global and update this state (which should be persisted)